File: www/Frame.js
var argscheck = require('cordova/argscheck'),
    channel = require('cordova/channel'),
    utils = require('cordova/utils'),
    exec = require('cordova/exec'),
    cordova = require('cordova');

function Frame() {
};

Frame.prototype.getHeader = function() {
	return this.header;
};

Frame.prototype.getSlider = function() {
	return this.slider;
};

Frame.prototype.getContent = function() {
	return this.content;
};

Frame.prototype.getMainColor = function(callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameGetMainColor", []);
};

Frame.prototype.setMainColor = function (mainColor, callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameSetMainColor", [mainColor]);
};

Frame.prototype.getMainColorDark = function(callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameGetMainColorDark", []);
};

Frame.prototype.setMainColorDark = function(mainColorDark, callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameSetMainColorDark", [mainColorDark]);
};

Frame.prototype.showSpinner = function(callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameShowSpinner", []);
};

Frame.prototype.hideSpinner = function(callback) {
	if (!callback) callback = function() {};
	exec(callback, function(){}, "Tgw", "frameHideSpinner", []);
};

module.exports = Frame;

File: src/android/Frame.java
package com.breixoam.tgw;

import org.apache.cordova.CordovaActivity;
import android.util.Log;
import android.graphics.Color;
import android.view.Window;
import android.app.ActivityManager;
import android.content.Context;
import android.support.v7.widget.Toolbar;

public class Frame {

    private static final String TAG = "TGWDebug Frame.java";

	protected String mainColor;
    protected String mainColorDark;
    protected Header header;
    protected Slider slider;
    protected Content content;

    /**
     * Constructor 
     */
    public Frame() {
        this.header = new Header();
        this.slider = new Slider();
        this.content = new Content();
    }

    //Return header
    public Header getHeader() {
        return this.header;
    }

    //Return slider
    public Slider getSlider() {
        return this.slider;
    }

    //Return content
    public Content getContent() {
        return this.content;
    }

    //Return main color
    public String getMainColor() {
    	return this.mainColor;
    }

    //Set and update de main color
    public void setMainColor(final String mainColor) {

    	this.mainColor = mainColor;
        final CordovaActivity activity = Tgw.activity;

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {

                ((TgwActivity) activity).toolBar.setBackgroundColor(Color.parseColor(mainColor));

                int currentapiVersion = android.os.Build.VERSION.SDK_INT;
                if (currentapiVersion >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                    ActivityManager activityManager = (ActivityManager) activity.getSystemService(Context.ACTIVITY_SERVICE);
                    for (ActivityManager.AppTask appTask : activityManager.getAppTasks()) {
                        if (appTask.getTaskInfo().id == activity.getTaskId()) {
                            ActivityManager.TaskDescription description = appTask.getTaskInfo().taskDescription;
                            ActivityManager.TaskDescription a = new ActivityManager.TaskDescription(description.getLabel(), description.getIcon(), Color.parseColor(mainColor));
                            activity.setTaskDescription(a);
                        }
                    }
                }

            }
        });
    }

    //Return main color dark
	public String getMainColorDark() {
    	return this.mainColorDark;
    }

    //Set and update de main color dark
    public void setMainColorDark(final String mainColorDark) {

    	this.mainColorDark = mainColorDark;
        final CordovaActivity activity = Tgw.activity;

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {

                int currentapiVersion = android.os.Build.VERSION.SDK_INT;
                if (currentapiVersion >= android.os.Build.VERSION_CODES.LOLLIPOP) {
                    final Window window = activity.getWindow();
                    window.setStatusBarColor(Color.parseColor(mainColorDark));
                }

            }
        });
    }

    //Show spinner dialog
    public void showSpinner() {

        final CordovaActivity activity = Tgw.activity;

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                ((TgwActivity) activity).progress.setMessage("Loading...");
                ((TgwActivity) activity).progress.show();
            }
        });

    }

    //Hide spinner dialog
    public void hideSpinner() {

        final CordovaActivity activity = Tgw.activity;

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                ((TgwActivity) activity).progress.dismiss();
            }
        });

    }

}
